[Docs] Command Syntax and Parameter fixes - 1

seaniepie opened the pull request:
Choose, Constant, Copy, Decrypt, Decrypt RSA, Do, Encrypt, Play, Play Video, revXMLRPC_SetMethod & revZipSetProgressCallback commands:
Fixes to Syntax and Parameter sections (plus minor formatting issues)

seaniepie committed:
[Docs] choose command - Fix Param formatting

Fix the formatting of the Parameters section. This also fixes a parsing error in TravisCL which assumed 'can' and 'rect' to be parameters to be found in the syntax.

seaniepie committed:
[docs] constant command - Fix Param formatting

- Move the word 'commas:' over to avoid it being wrongly recognised as a Parameter by Travis
- Add naming convention info to notes in the description 
- Change constant names in examples to include the prefix k

seaniepie committed:
[Docs] copy command - Fix Param formatting

- Added curly brackets around the variants in the Syntax
- Descriptions added to empty parameters, chunk and field

seaniepie committed:
[Docs] decrypt command - Fix Syntax variant

Curly brackets put around variant instead of square brackets

seaniepie committed:
[Docs] encrypt command - Fix Syntax variant

Replaced square brackets for curly brackets around variant.

seaniepie committed:
[Docs] decrypt RSA command - Fix Syntax variant

- Removed reference to the unnecessary 'source' parameter in favour of the 'message' param.
- Added a summary (required)
- Put example scripts into inline blocks

seaniepie committed:
[Docs] decrypt RSA command - Fix Syntax and Params

- Fix 'alternateLanguageName:' in syntax and remove anomaly in Parameters

seaniepie committed:
[Docs] play command - Fix Syntax and Params

- Removed unused square bracket
- Reformatted the end note

seaniepie committed:
[Docs] play-video command - Fix Syntax and Params

- Replaced parenthesis with curly brackets in Syntax
- Added tag in last paragraph

seaniepie committed:
[Docs] play-video command - Fix Params

- Subset of callback parameters listed within table

seaniepie committed:
[Docs] play-video command - Fix Syntax & Params

Parameter names made to be accepted by both the dictionary display and by TravisCL by adding in underscores

montegoulding commented on docs/dictionary/command/choose.lcdoc:
```diff
@@ -22,16 +22,20 @@ choose tSavedTool tool
+    * curve: draw a curved line in an image 
+    * regular polygon: draw a regular polygon shape in an image 
+    * dropper: pick up a color from an image
 -   "browse": perform actions such as clicking buttons and typing in
```
@seaniepie the above should be formatted like this one

seaniepie replied:
Like which one?

seaniepie replied:
Do you mean with '-  pencil: draw' rather than '\* pencil: draw'

livecodeali replied:
Yeah they should all be `- "option": description of option`

montegoulding commented on docs/dictionary/command/constant.lcdoc:
```diff
@@ -14,15 +14,15 @@ OS: mac, windows, linux, ios, android
+constant kEntryA="EF9993333WX786",kEntryB="GJ773281YX342"
 
 Parameters:
 constantsList (enum):
```
It's not an enum so we should delete `(enum)` here

montegoulding commented on docs/dictionary/command/decrypt-using-rsa.lcdoc:
```diff
@@ -36,12 +37,12 @@ Use the <decrypt using rsa> command to decrypt a message using RSA
 
 Use the form
-decrypt <message> with private key <key> 
+'decrypt <message> with private key <key> '
```
This should have an empty line above and below and be indented 4 spaces so it is code formatted. Perhaps a colon after form too.

montegoulding commented on docs/dictionary/command/decrypt-using-rsa.lcdoc:
```diff
@@ -36,12 +37,12 @@ Use the <decrypt using rsa> command to decrypt a message using RSA
 
 Use the form
-decrypt <message> with public key <key> 
+'decrypt <message> with public key <key>' 
```
As above

seaniepie committed:
Lose enum and put angle brackets around sub-params

seaniepie commented on docs/dictionary/command/constant.lcdoc:
```diff
@@ -14,18 +14,18 @@ OS: mac, windows, linux, ios, android
-The constantsList consists of one or more name=value pairs, separated by
-commas: 
+constantsList:
+The constantsList consists of one or more <name>=<value> pairs, 
```
Do the angle brackets make more sense here?

seaniepie replied:
They don't work

livecodeali replied:
As they aren't strictly parameters or links, the angle brackets won't work. You could do it with `&lt;` and `&gt;` though. I think we quite often use angle brackets for placeholders in the docs, so I think the change does make sense.

seaniepie committed:
Put single quotes around sub parameters

seaniepie committed:
Reformat description

- The two paragraphs associated with each Form example have been consolidated to make clearer sense.
- The 'Generating Key Pairs' part has been formatted as a note

seaniepie committed:
Standardise the bullets in the parameters table

- Changed all to hyphen
- Removed double quotes from some toolNames

livecodeali commented on docs/dictionary/command/revXMLRPC_SetMethod.lcdoc:
```diff
@@ -2,7 +2,7 @@ Name: revXMLRPC_SetMethod
 Type: command
 
-Syntax: revXMLRPC_SetMethod <XML-RPC document>,<method name>
+Syntax: revXMLRPC_SetMethod <XML_RPC_documentID>,<method_name>
```
Could you do these as camel case instead? It probably makes sense to keep that convention throughout the docs.

seaniepie committed:
Parameters set to camelCase

livecodeali commented:
@livecode-vulcan review ok b1df1d4

livecode-vulcan commented:
💙 review by @livecodeali ok b1df1d4cd08aea68142697f68c7a8d3dc967ff64

livecode-vulcan commented:
😎 test success b1df1d4cd08aea68142697f68c7a8d3dc967ff64
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/403)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/408)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/403)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/403)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/409)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/403)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/403)
